### PR TITLE
Refactoring ClassHelper for readability.

### PR DESCRIPTION
**Pull request by krmahadevan**
Also removed getEnclosingClass() which is now
available in JDK itself.

Fixes # .

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/internal/ClassHelper.java [hunk 1]
```diff
@@ -458,22 +392,81 @@ public static Object createInstance1(
...
+        enclosingIClass.addInstance(o);
+        enclosingInstances = new Object[]{o};
       }
+    } else {
+      enclosingInstances = new Object[]{ec.newInstance()};
     }
+    Object enclosingClassInstance = enclosingInstances[0];
 
-    return result;
+    return new Object[]{enclosingClassInstance};
```
Comment (juherr) [inline src/main/java/org/testng/internal/ClassHelper.java]: Why not returning an object as it is the intent of the method?

```java
if (null == enclosingIClass) {
  return ec.newInstance();
}
Object[] enclosingInstances = enclosingIClass.getInstances(false);
if (enclosingInstances != null && enclosingInstances.length > 0) {
  return enclosingInstances[0];
}

Object o = factory.newInstance(ec.getConstructor(ec));
enclosingIClass.addInstance(o);
return o;
```

Comment (juherr) [reply src/main/java/org/testng/internal/ClassHelper.java]: And adapt to where you use it:

```java
parameters = new Object[]{computeParameters(classes, ec, factory)};
```

**Review by juherr — CHANGES_REQUESTED** [review]
Good refactoring 👍 Just 2 little remarks.

File: src/main/java/org/testng/internal/ClassHelper.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/internal/ClassHelper.java]: @juherr - That was a good one. I didn't see hard enough there... I have fixed it, amended the commit and pushed it. Can you please check now

**Review by juherr — APPROVED** [review]
LGTM 👍 

Ping @cbeust

**Comment by krmahadevan** [comment]
ping @cbeust - Can you please help with getting this merged ?

**Pull request merged**